testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/ejb/transaction/bmt/timeout/StatelessTimeoutTestCase.java

```diff
@@ -0,0 +1,123 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.test.integration.ejb.transaction.bmt.timeout;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.as.test.integration.transactions.TransactionTestLookupUtil;
+import org.jboss.as.test.integration.transactions.TransactionCheckerSingleton;
+import org.jboss.as.test.integration.transactions.TxTestUtil;
+import org.jboss.as.test.shared.TimeoutUtil;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import javax.inject.Inject;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+
+/**
+ * BMT test where transaction timeout is involved.
+ * The bean uses mock XAResources to simulate to get 2PC happen.
+ */
+@RunWith(Arquillian.class)
+public class StatelessTimeoutTestCase {
+
+    @ArquillianResource
+    private InitialContext initCtx;
+
+    @Inject
+    private TransactionCheckerSingleton checker;
+
+    @Deployment
+    public static Archive<?> createDeployment() {
+        final JavaArchive jar = ShrinkWrap.create(JavaArchive.class, "stateless-btm-txn-timeout.jar")
+            .addPackage(StatelessTimeoutTestCase.class.getPackage())
+            .addPackage(TxTestUtil.class.getPackage())
+            .addClasses(TimeoutUtil.class)
+            .addAsManifestResource(new StringAsset("<beans></beans>"),  "beans.xml");
+        return jar;
+    }
+
+    @Before
+    public void startUp() throws NamingException {
+        checker.resetAll();
+    }
+
+    /**
+     * Calling BTM method where transaction timeout is not modified.
+     * Call is expected to be processed.
+     */
+    @Test
+    public void noTimeout() throws Exception {
+        StatelessBmtBean bean = TransactionTestLookupUtil.lookupModule(initCtx, StatelessBmtBean.class);
+        bean.testTransaction(1, 0);
+    }
+
+    /**
+     * Calling BTM method where 3 transactions are started and one of them timeouts.
+     * Each transaction contains 2 XA Resources.
+     * <br>
+     * The assertion checks that 2 transactions were committed and one rolled-back.
+     */
+    @Test
+    public void timeout() throws Exception {
+        StatelessBmtBean bean = TransactionTestLookupUtil.lookupModule(initCtx, StatelessBmtBean.class);
```

Comment:
These tests seem like a fundamental test of the transaction manager, and don't appear to be EJB specific. I don't really know if these are needed, as I assume this would be tested fairly comprehensively by the Narayana test suite.

@tomjenkinson do you think it is worth having these tests? They basically just test that a transaction than times out is not committed.